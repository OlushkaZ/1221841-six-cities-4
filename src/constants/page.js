export const PageType = {
  MAIN: `main`,
  DETAILS: `details`,
};

export const CardClasses = {
  MAIN: {
    card: `cities__place-`,
    wrapper: `cities`,
    cards: `cities__places-`,
    map: `cities`,
  },
  PROPERTY: {
    card: `near-places__`,
    wrapper: `near-places`,
    cards: `near-places__`,
    map: `property`
  }
};
