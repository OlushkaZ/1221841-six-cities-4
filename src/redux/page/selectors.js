import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.PAGE;

export const getPropertyOffer = (state) => {
  return state[NAME_SPACE].propertyOffer;
};

export const getPageStep = (state) => {
  return state[NAME_SPACE].step;
};

export const getActiveOffer = (state) => {
  return state[NAME_SPACE].activeOffer;
};

export const getActiveCity = (state) => {
  return state[NAME_SPACE].city;
};


