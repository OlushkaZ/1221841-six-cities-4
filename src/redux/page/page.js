import {PageType} from "../../constants/page.js";
const initialState = {
  city: null,
  step: PageType.MAIN,
  propertyOffer: null,
  activeOffer: null,
};

const ActionType = {
  GET_OFFERS: `GET_OFFERS`,
  CHOOSE_CITY: `CHOOSE_CITY`,
  CHANGE_PAGE_TYPE: `CHANGE_PAGE_TYPE`,
  MAKE_OFFER_ACTIVE: `MAKE_OFFER_ACTIVE`,
  MAKE_OFFER_INACTIVE: `MAKE_CARD_INACTIVE`,
};

const ActionCreator = {
  chooseCity: (activeCity) => {
    return (
      {
        type: ActionType.CHOOSE_CITY,
        city: activeCity,
      }
    );
  },

  changePageType: (offer) => {
    return (
      {
        type: ActionType.CHANGE_PAGE_TYPE,
        step: PageType.DETAILS,
        propertyOffer: offer,
      }
    );
  },

  makeOfferCardActive: (offer) => {
    return (
      {
        type: ActionType.MAKE_OFFER_ACTIVE,
        activeOffer: offer,
      }
    );
  },

  makeOfferInactive: () => {
    return (
      {
        type: ActionType.MAKE_OFFER_INACTIVE,
        activeOffer: null,
      }
    );
  },

};


const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHOOSE_CITY:
      const {city, offers: filteredOffers} = action;
      return Object.assign({}, state, {
        city,
        offers: filteredOffers,
      });

    case ActionType.CHANGE_PAGE_TYPE:
      const {step, propertyOffer} = action;
      return Object.assign({}, state, {
        step,
        propertyOffer,
      });

    case ActionType.MAKE_OFFER_ACTIVE:
      const {activeOffer} = action;
      return Object.assign({}, state, {
        activeOffer,
      });

    case ActionType.MAKE_OFFER_INACTIVE:
      const {activeOffer: inactiveOffer} = action;
      return Object.assign({}, state, {
        activeOffer: inactiveOffer,
      });
  }

  return state;
};

export {reducer, ActionType, ActionCreator};
