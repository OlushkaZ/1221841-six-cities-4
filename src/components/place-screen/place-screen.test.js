import React from "react";
import renderer from "react-test-renderer";
import PlaceScreen from "./place-screen.jsx";
import {PageType} from "../../constants/page.js";

const props = [
  {
    children: <div className="children-component" />,
    color: `gray`,
    type: PageType.MAIN,
    authorizationStatus: `NO_AUTH`,
  },
  {
    children: <div className="children-component" />,
    type: PageType.DETAILS,
    authorizationStatus: `NO_AUTH`,
  },
];

describe(`PlaceScreen render correctly`, () => {
  it(`with Main`, () => {
    const tree = renderer
    .create(
        <PlaceScreen
          {...props[0]}/>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with Place Property`, () => {
    const tree = renderer
    .create(
        <PlaceScreen
          {...props[1]}/>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
