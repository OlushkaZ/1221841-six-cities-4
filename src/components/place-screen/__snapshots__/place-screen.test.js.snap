// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceScreen render correctly with Main 1`] = `
<div
  className="page
      page--gray
      page--main
    "
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <div
    className="children-component"
  />
</div>
`;

exports[`PlaceScreen render correctly with Place Property 1`] = `
<div
  className="page
      
      page--details
    "
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <div
    className="children-component"
  />
</div>
`;
